File: Day299/uniqueLength3PalindromicSubsequences.go
func countPalindromicSubsequence(s string) int {
    answer :=0 
    // l := len(s)
    
    for c:= 'a'; c<='z'; c++ {
        first := strings.Index(s, string(c))
        last := strings.LastIndex(s, string(c))
        if first!= -1 && last !=-1  && first < last {
            set := make(map[byte]bool)
            for i:=first+1; i<last; i++ {
                set[s[i]] =  true
            }
            answer += len(set)
        }
    }
    return answer
}

File: Day299/uniqueLength3PalindromicSubsequences.java
class Solution {
    public int countPalindromicSubsequence(String s) {
        int answer  = 0;
        for (char c='a'; c<='z'; c++) {
            int first = s.indexOf(c);
            int last = s.lastIndexOf(c);
            if (first != -1 && last != -1 && first < last) {
                Set<Character> set = new HashSet<>();
                for (int i=first+1; i<last; i++) {
                    set.add(s.charAt(i));
                } 
                answer += set.size();
            }
        }
        return answer;
    }
}